lib/exprotobuf/define_message.ex

```diff
@@ -104,17 +104,20 @@ defmodule Protobuf.DefineMessage do
   end
 
   defp record_fields(fields) do
-    for %Field{name: name, occurrence: occurrence} <- fields do
-      {name, case occurrence do
-        :repeated -> []
-        _ -> nil
-      end}
-    end
-    ++
-    for %OneOfField{name: name} <- fields do
-      {name, nil}
-    end
-
+    fields
+    |> Enum.map(fn(field) ->
+      case field do
+        %Field{name: name, occurrence: :repeated} ->
+          {name, []}
+        %Field{name: name} ->
+          {name, nil}
+        %OneOfField{name: name} ->
+          {name, nil}
+        _ ->
+          nil
+      end
+    end)
+    |> Enum.reject(fn(field) -> is_nil(field) end)
```

Comment:
This is just so the default value for a repeated field is an empty list, rather than nil. It's possible it's not necessary, but I think it's a sane choice, so let's keep it for now.